redisearch/client.py

```diff
@@ -102,9 +108,17 @@ class IndexDefinition(object):
     SCORE = 'SCORE'
     PAYLOAD_FIELD = 'PAYLOAD_FIELD'
         
-    def __init__(self, prefix=[], filter=None, language_field=None, language=None, score_field=None, score=1.0, payload_field=None):
-
-        args = [self.ON, self.HASH]
+    def __init__(self, prefix=[], filter=None, language_field=None, language=None, score_field=None, score=1.0, payload_field=None, index_type=None):
+        args = []
+
+        if index_type:
+            args.append(self.ON)
+            if index_type is IndexType.HASH:
```

Comment:
Instead, I'd do:

if index_type is IndexType.JSON:
    #args.append for json
elif index_type is IndexType.HASH:
    # args for hash
elif index_type is not None:
    raise RunTimeError(....)